jax/core.py

```diff
@@ -1343,6 +1343,8 @@ def write(v: Var, a: AbstractValue) -> None:
     prim = eqn.primitive
     try:
       in_avals = map(read, eqn.invars)
+      typecheck_assert(all(not isinstance(ina, ConcreteArray) for ina in in_avals),
+                       f"Primitive {prim} takes ConcreteArray abstract inputs")
```

Comment:
```suggestion
                       f"Equation given ConcreteArray type inputs")
```

Minor: the primitive name and equation are already included as context whenever `typecheck_assert` raises in this block (see a few lines down in `except ...`)